### PR TITLE
fix dropout prob

### PR DESCRIPTION
**Pull request by avdmitry**
Hello, Andrej @karpathy 

I think here should be (1-this.drop_prob). Let's consider probability = 0.1 - we should drop 10% and after that scale weights only slightly onto 0.9.

WBR,
Dmitriy

### CODE REVIEW

**Comment by jruales** [comment]
E[weight] = drop_prob*0 + (1-drop_prob)*w[i] = (1-drop_prob)*w[i], so I agree with @avdmitry

**Comment by jruales** [comment]
@avdmitry 's solution agrees with Figure 2 in the third page of the dropout paper: http://www.cs.toronto.edu/~rsalakhu/papers/srivastava14a.pdf

The Keras library takes a different approach (but still somewhat equivalent): it divides by the retain probability during training time so that during testing the dropout layer is just an identity function
https://github.com/fchollet/keras/blob/master/keras/layers/core.py#L621
https://github.com/fchollet/keras/blob/master/keras/backend/theano_backend.py#L555
